src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rest/QueryActionRequestHandler.java

```diff
@@ -0,0 +1,139 @@
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence.actions.PersistedAction;
+import com.google.gson.JsonObject;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.security.InvalidParameterException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.apache.logging.log4j.util.Supplier;
+
+public class QueryActionRequestHandler extends MetricsHandler implements HttpHandler {
+
+    private static final Logger LOG = LogManager.getLogger(QueryActionRequestHandler.class);
+    private Persistable persistable;
+    private AppContext appContext;
+
+    public QueryActionRequestHandler(final AppContext appContext) {
+        this.appContext = appContext;
+    }
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+        String requestMethod = exchange.getRequestMethod();
+
+        if (requestMethod.equalsIgnoreCase("GET")) {
+            LOG.debug("Action Query handler called.");
+            exchange.getResponseHeaders().set("Content-Type", "application/json");
+
+            try {
+                synchronized (this) {
+                    String query = exchange.getRequestURI().getQuery();
+                    handleActionRequest(exchange);
+                }
+            } catch (InvalidParameterException e) {
+                LOG.error(
+                        (Supplier<?>)
+                                () ->
+                                        new ParameterizedMessage(
+                                                "QueryException {} ExceptionCode: {}.",
+                                                e.toString(),
+                                                StatExceptionCode.REQUEST_ERROR.toString()),
+                        e);
+                String response = "{\"error\":\"" + e.getMessage() + "\"}";
+                sendResponse(exchange, response, HttpURLConnection.HTTP_BAD_REQUEST);
+            } catch (Exception e) {
+                LOG.error(
+                        (Supplier<?>)
+                                () ->
+                                        new ParameterizedMessage(
+                                                "QueryException {} ExceptionCode: {}.",
+                                                e.toString(),
+                                                StatExceptionCode.REQUEST_ERROR.toString()),
+                        e);
+                String response = "{\"error\":\"" + e.toString() + "\"}";
+                sendResponse(exchange, response, HttpURLConnection.HTTP_INTERNAL_ERROR);
+            }
+        } else {
+            exchange.sendResponseHeaders(HttpURLConnection.HTTP_NOT_FOUND, -1);
+            exchange.close();
```

Comment:
This `exchange.close()` should be outside the `else {}` so that we close the object for all requests. Please change this for `QueryRcaRequestHandler` as well.

Reply:
done